Refresh of the Channels page, introducing branches

caldav opened the pull request:
* Clean up and simplify existing content 
* Introduce branches
* Add syntax/structure section
* New examples
* Page renamed back to "Channels"

Hey @facundobatista, @evandandrea and @Roadmaster, please have a look at these changes.

## Building the doc

To build the documentation and view it inside snapcraft.io, you need to build snapcraft.io and change the path where docs are imported from. The following snippet will take care of it, and serve the build on port 8004.

``` bash
git clone git@github.com:ubuntudesign/snapcraft.io.git
git clone git@github.com:CanonicalLtd/snappy-docs.git
cd snappy-docs
git fetch origin
git checkout origin/channels-page-refresh
cd ../snapcraft.io
sed -i "s/https:\/\/github\.com\/CanonicalLtd\/snappy-docs\.git/\.\.\/snappy-docs/g" run
./run

```

Then, point your browser to http://0.0.0.0:8004/docs/reference/channels

sergiusens commented on reference/channels.md:
```diff
@@ -1,92 +1,196 @@
 ---
 
-The tracks and channels mechanism provided by the store allow you to distribute multiple versions of the same snap, allowing users to follow and get automated upgrades for daily releases (published manually or automatically [through Launchpad or Travis](/docs/build-snaps/ci-integration)), beta, candidate and stable releases. It also provides a way for your users to stick to a specific software se ...
+Release channels allow you to distribute multiple versions of a snap at the same time.
```
I think it might be better communicated in usefulness if this added the idea of subscription.

Channels allow users to subscribe to different release cadences for a snap, be it by major version bumps or in-development releases.

Or something like that.

Roadmaster commented on reference/channels.md:
```diff
@@ -1,92 +1,196 @@
+## Track
 
-## Channels
+Tracks allow you to publish different series of your software (1.0, 2.0, etc.) and let users follow and get automated upgrades from specific one.
```
either "a specific one" or "specific ones".

sparkiegeek commented on reference/channels.md:
```diff
@@ -1,92 +1,196 @@
 
        $ snap install my-app --channel=1.0/beta
+
+* Installing "my-app" from the fix-for-bug123 branch on top of the "stable" level of the "latest" track:
```
s/on top /

the branch is *of* the latest/stable snap

sparkiegeek commented on reference/channels.md:
```diff
@@ -1,92 +1,196 @@
+  Channel syntax: `latest/stable/fix-for-bug123`
+
+         $ snap install my-app --channel=stable/fix-for-bug123
+      Or $ snap install --channel=latest/stable/fix-for-bug123
```
needs `my-app` at the end

caldav commented:
Thanks @sergiusens @Roadmaster @sparkiegeek , I believe I fixed them all.

@sergiusens I've changed the beginning with something very close to what you suggested, and altered the wording of the next paragraph to address it more directly to developers.

sparkiegeek commented:
:+1:

Roadmaster commented:
+1 from me, thanks!
